spec/lib/granite/action_spec.rb

```diff
@@ -7,6 +7,55 @@
     end
   end
 
+  describe 'Dependency Injection' do
+    before do
+      stub_class(:action, Granite::Action) do
+        allow_if { true }
+
+        attribute :comment, String
+
+        attr_reader :my_dep, :another_dep
+        private :my_dep, :another_dep
+
+        def initialize(*args, my_dep: Hash[a: 1], another_dep: 'Foo', **kwargs, &block)
+          @my_dep = my_dep
+          @another_dep = another_dep
+          super(*args, **kwargs, &block)
+        end
+      end
+
+      stub_class(:another_action, Granite::Action) do
+        allow_if { true }
+      end
+    end
+
+    it 'creates private getter' do
+      action = Action.new(comment: 'blah blah blah')
+      expect { action.my_dep }.to raise_error(NoMethodError, /private method `my_dep' called/)
+    end
+
+    it 'uses default value' do
```

Comment:
Again you're testing ruby

Reply:
@andriusch I'm testing that attributes would NOT interfere with the basics of Ruby.

Reply:
Removed by https://github.com/toptal/granite/pull/64/commits/2a98b4881927be094aac10d5ec3d882105bdba3d